ACLWeb.js

```diff
@@ -1,184 +1,196 @@
-		results = [];
-		doImport = false;
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU Affero General Public License for more details.
 
-		var bibFileNode = bibFileNodes.iterateNext();
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero.  If not, see <http://www.gnu.org/licenses/>.
 
-		while (bibFileNode) {
-			var bibFileName = bibFileNode.getAttribute("href");
-			var bibFile = bibFileName.substring(0, bibFileName.length-4);
+	***** END LICENSE BLOCK *****
+*/
 
-			var bNodes = doc.evaluate('//a[@href="' + bibFileName + '"]/following-sibling::b[position()=1]', doc, null, XPathResult.ANY_TYPE, null);	// These nodes contain author information
-
-			// Extract authors' last names
-			var authorLasts = new Array();
+function detectWeb(doc, url) {
+	if (doc.contentType === 'application/pdf' || url.endsWith('.bib')) {
+		return 'conferencePaper';
+	} else {
+		return 'multiple';
+	}
+}
 
-			var bNode = bNodes.iterateNext();
-			var authorsS = bNode.innerHTML;	// may include markup: potentially <author>, <first>, <von>, and/or <last> tags
-			authorsS = authorsS.replace(/[<][/]?author[>]/g, "");
-			var authors = authorsS.split("; ");
-			for (var a in authors) {
-				var authorS = authors[a];
-				var m = authorS.match(/[<]von[>]([^<]+)[<][/]von[>]/);
-				var last = "";
-				if (m!=null)	// we expect there is a <last> tag if there is a <von> tag
-					last = m[1] + " ";
-				m = authorS.match(/[<]last[>]([^<]+)[<][/]last[>]/);
-				if (m!=null)
-					last += m[1];
-				else {
-					var name = authorS.replace(/[<][^>]+[>]/g, "");	// remove all markup
-					if (name=="Entire volume")
-						last = name;
-					else {
-						var parts = name.split(" ");
-						last = parts[parts.length-1];
-						if (parts.length>1) {
-							var penultInitial = parts[parts.length-2].substr(0,1);
-							if (penultInitial.toUpperCase()!=penultInitial)	// e.g. van Dyke
-								last = name[parts.length-2] + " " + last;
-						}
-					}
+function doWeb(doc, url) {
+	if (detectWeb(doc, url) === 'multiple') {
+		Zotero.selectItems(extractFullProceedings(doc), function (selected) {
+			if (!selected) {
+				return true;
+			}
+			
+			Object.keys(selected).forEach(function (id) {
+				let bibtexElement = ZU.xpath(doc, '//a[contains(@href, "' + id + '.bib")]');
+				
+				// Sometimes there won't be a BibTeX link, so we need to check
+				// and scrape directly from the proceedings page if there isn't.
+				if (bibtexElement.length) {
+					let bibtexURL = bibtexElement[0].href;
+					ZU.doGet(bibtexURL, function(responseString, responseObj, url) {
+						scrapeBibtex(responseString, url);
+					});
+				} else {
+					scrapeProceedings(doc, id);
 				}
-				authorLasts.push(last);
+			});
+		});
+	} else if(url.endsWith('.bib')) {
+		// e.g. http://www.aclweb.org/anthology/P10-4014.bib
```

Comment:
This looks like a good test case we also should add.

Reply:
This test fails with an error

```
21:58:35 TranslatorTester: Running ACL Test 3
21:58:35 TranslatorTester: Translating http://www.aclweb.org/anthology/P10-4014.bib
21:58:35 TranslatorTester: ACL Test 3: failed (Translation failed to initialize: TypeError: doc is null)
```

I have double checked that it works in Chrome though, is that sufficient?

Reply:
Ah, okay. Yes, fine then.